nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1959 @@
+				},
+				"transport_zone_profile_id": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Identifiers of the transport zone profiles associated with this transport zone endpoint on this transport node",
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getUplinksSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"uplink_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Uplink name from UplinkHostSwitch profile",
+				},
+				"vds_lag_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Link Aggregation Group (LAG) name of Virtual Distributed Switch",
+				},
+				"vds_uplink_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Uplink name of VMware vSphere Distributed Switch (VDS)",
+				},
+			},
+		},
+	}
+}
+
+func getHostSwitchProfileIDsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Identifiers of host switch profiles to be associated with this host switch",
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"key": {
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(hostSwitchProfileIDKeyValues, false),
+					Required:     true,
+				},
+				"value": {
+					Type:        schema.TypeString,
+					Description: "key value",
+					Required:    true,
+				},
+			},
+		},
+	}
+}
+
+func getIPAssignmentSchema(varPath string) *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Specification for IPs to be used with host switch virtual tunnel endpoints",
+		MaxItems:    1,
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"assigned_by_dhcp": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					//TODO: as ExactlyOneOf cannot be used, use some alternate validation
+					//ExactlyOneOf: []string{
+					//	varPath + ".0.static_ip_list_spec",
+					//	varPath + ".0.static_ip_mac_list_spec",
+					//	varPath + ".0.static_ip_pool_spec",
+					//},
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:        schema.TypeBool,
+								Required:    true,
+								Description: "Enables DHCP assignment. Should be set to true",
+							},
+						},
+					},
+				},
+				"static_ip_list_spec": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_list": {
```

Comment:
we normally call list of IPs `ip_addresses` in the provider, lets do the same for consistency here?